webapp/src/js/app.js

```diff
@@ -54,6 +56,27 @@ var _ = require('underscore');
 _.templateSettings = {
   interpolate: /\{\{(.+?)\}\}/g,
 };
+var reduxLoggerConfig = {
+  actionTransformer: function(action) {
+    var updatedAction = {};
+    try {
+      JSON.stringify(action);
+    } catch(error) {
+      updatedAction.payload = 'Payload is not serializable';
+    }
+    return Object.assign({}, action, updatedAction);
+  },
+  stateTransformer: function(state) {
+    var updatedState = {};
+    try {
+      JSON.stringify(state.selected);
+    } catch(error) {
+      updatedState.selected = 'Selected is not serializable';
+    }
+    return Object.assign({}, state, updatedState);
+  },
```

Comment:
Because some docs are recursive (contact.parent.contact.parent...), redux-logger cannot log any state or actions containing these docs. For now I've just opted to not log them, but it would be neat to come up with a better solution for this as being able to see these state changes is really useful for debugging.

Reply:
Yeah the general thinking and theory around redux is to try to keep the store as flat as possible with minimal deep structures, so I think we're in the wrong here for having recursive structures unfortunately.

No that seems like a sensible solution! That's in the lineage library right? Should be pretty straightforward to add in here I'll give it a go

Reply:
Fixed 👍